pkg/regionpicker/gcp.go

```diff
@@ -0,0 +1,17 @@
+package regionpicker
+
+const (
+	CloudGCP CloudName = "gcp"
+)
+
+type GCP struct{}
+
+// Regions returns a list of all known-gcp regions. This should not be used
+// over native GCP region fetching.
+// IDEA: One day we could probably get this from GCP. That is authenticated though.
+func (*GCP) Regions() []region {
```

Comment:
I suggest returning a strongly typed exported `Regions` rather than unexported `[]region`.  This would allow you to just move your `best(regions)` helper as a method on `Regions`:

```golang
type Regions []region
func (r Regions) Nearest(ctx context.Context) (RegionName, error) { .... }
```

This would also make your main `FindBest` function read a lot better: `cloud.Regions().Nearest()`

Reply:
I didn't really want to export `region` because I wanted to protect against usage of this outside of tooling, that being said.... I'm not sure if that's a fair goal to have. So, I'm not super against exporting this then.

Reply:
You don't need to expose region -- just Regions and RegionName :)